src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCaller.java

```diff
@@ -209,15 +209,11 @@ public void onTraversalStart() {
 
     @Override
     public void apply(final AssemblyRegion region, final ReferenceContext referenceContext, final FeatureContext featureContext ) {
```

Comment:
(Thinking out loud here)

`apply()` is a one-liner now so duplication is no longer as big of an issue, but it is still a little unsatisfying. I wonder if HC and M2 should share code via an interface with default methods (a mixin?). But then how's that different from having a util class? 

I will come back to this after I go through the PR.